dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandler.kt

```diff
@@ -7,26 +7,34 @@
 package com.datadog.android.log.internal.logger
 
 import android.util.Log as AndroidLog
+import com.datadog.android.core.internal.CoreFeature
 import com.datadog.android.core.internal.data.Writer
 import com.datadog.android.core.internal.net.info.NetworkInfoProvider
 import com.datadog.android.core.internal.sampling.RateBasedSampler
 import com.datadog.android.core.internal.sampling.Sampler
-import com.datadog.android.log.LogAttributes
 import com.datadog.android.log.internal.domain.Log
+import com.datadog.android.log.internal.domain.LogGenerator
 import com.datadog.android.log.internal.user.UserInfoProvider
 import com.datadog.android.rum.GlobalRum
 import com.datadog.android.rum.RumErrorSource
-import io.opentracing.util.GlobalTracer
 
 internal class DatadogLogHandler(
-    internal val serviceName: String,
-    internal val loggerName: String,
+    serviceName: String,
```

Comment:
I'd rather remove the `service`, `loggerName`, `networkInfoProvider` and `userInfoProvider` arguments and simply inject a `LogGenerator` directly.
I understand that doing this limits the impact of the change, but this makes this constructor bigger, and will make future changes even harder to do.

Reply:
That's the first approach I had actually but then it felt weird to create the `LogGenerator` in the `DatadogLogHandler` test, I wanted to make it invisible somehow but yeah injecting it makes sense to me too.

Reply:
You can simply mock it in the tests, you don't have to create it.

Reply:
I prefer creating it there, this way I will also keep the old tests in the `DatadogLogHandler`, I feel safer this way ;)